Makefile

```diff
@@ -48,35 +46,22 @@ install-docs: docs
 
 install: install-docs
 
-cross: vendor
-	GOOS=windows $(MAKE) build-internal BUILDTAGS="$(BUILDTAGS) $(BUILD_TAGS_WINDOWS_CROSS)"
-	GOOS=darwin $(MAKE) build-internal BUILDTAGS="$(BUILDTAGS) $(BUILD_TAGS_DARWIN_CROSS)"
+cross:
+	GOOS=windows $(MAKE) build BUILDTAGS="$(BUILDTAGS) $(BUILD_TAGS_WINDOWS_CROSS)"
+	GOOS=darwin $(MAKE) build BUILDTAGS="$(BUILDTAGS) $(BUILD_TAGS_DARWIN_CROSS)"
 
 tools: tools.timestamp
 
 tools.timestamp: Makefile
 	@go get -u $(BUILDFLAGS) golang.org/x/lint/golint
 	@go get $(BUILDFLAGS) github.com/vbatts/git-validation
-	@go get -u github.com/rancher/trash
 	@touch tools.timestamp
```

Comment:
Looking at the build logs:
```
go get: warning: modules disabled by GO111MODULE=auto in GOPATH/src;
	ignoring go.mod;
	see 'go help modules'
go get: warning: modules disabled by GO111MODULE=auto in GOPATH/src;
	ignoring go.mod;
	see 'go help modules'
git-validation -q -run DCO,short-subject,dangling-whitespace
```
It would be nice to clean that up; AFAICS it comes from the `tools.timestamp` rule.

---

More importantly, the `go get` commands above, when run outside of `GOPATH` actually add the three commands as dependencies in `go.mod`; that seems quite undesirable (consumers of the library neither need nor want to include our testing infrastructure).  Sadly, there isn’t an obvious way to bypass that (at least quickly skimming https://github.com/golang/go/issues/27643 and several of the linked discussions).

I guess the closest to working is the https://stackoverflow.com/questions/53368187/go-modules-installing-go-tools solution of using an unrelated temporary working directory (except that it should use `mktemp -d` instead of hard-coding a single shared=possibly malicious directory).

Reply:
I think adding an explicit `GO111MODULE=off` should fix it.

Reply:
For now; https://github.com/containers/skopeo/pull/695 says that Go 1.13 ignores that variable. But this is not my area of expertise.

Reply:
Oh great :(

Reply:
It looks like Go 1.13 only changes the behaviour of `GO111MODULE=auto` though:
https://github.com/golang/go/blob/ad4ed87f80c33f23bdd3767ef3208f15a1fb5c90/doc/go1.13.html#L181-L190

So explicitly setting `GO111MODULE=off` should still work with Go 1.13.

Reply:
Thanks, looking at the code `GO111MODULE=off` does seem to set at least the local flags in Go’s `src/cmd/go/internal/modload/init.go` and `src/go/build/build.go` to turn off modules, so this code should be fine.

Yet, `GO111MODULE=off` was exactly what Skopeo was using before https://github.com/containers/skopeo/pull/695 , so I don’t quite know what is going on. Hopefully @vrothberg can help.

Reply:
Starting with Go v1.13 the variable will be ignored (since go modules are declared stable) but we still need it for backwards compat with 1.11 and 1.12.

As @mtrmac has mentioned, any `go get` and other go commands can potentially alter the module, so we need to be careful. What we do in buildah, for instance, is to keep the CI dependencies in a separate (sub) go module: https://github.com/containers/buildah/tree/master/tests/tools

What we do there is have a dedicated go module containing the tools for the CI that will be compiled on-demand. The code could also be downloaded on-demand, if desired. To trick go into believing we have dependencies on the tools, we have a dummy `.go` file with `import _ "github.com/.../..."` lines (https://github.com/containers/buildah/blob/master/tests/tools/tools.go).

I can take care of doing that once this PR is merged.

Reply:
> Starting with Go v1.13 the variable will be ignored (since go modules are declared stable) but we still need it for backwards compat with 1.11 and 1.12.

That’s not what the code actually does AFAICT. There are quite a few checks for `os.Getenv("GO111MODULE") == "off"` in the codebase, even on `master`. Or have they dropped that check from some important place that affects these projects?

Reply:
Thanks for clarifying. I didn't actually check the go code but based my statement on the various build issues in Fedora when flipping it to v1.13. So it seems that `GO111MODULE` can still be turned off explicitly ... although it did not help with `go build` apparently.